pkg/controller/common/helpers.go

```diff
@@ -125,13 +125,31 @@ func WriteTerminationError(err error) {
 	glog.Fatal(msg)
 }
 
+func replaceOrAppend(files []ign2types.File, file ign2types.File) {
+	for i, f := range files {
+		if f.Node.Path == file.Node.Path {
+			files[i] = file
+			return
+		}
+	}
+	files = append(files, file)
```

Comment:
I think we're missing a `return` after this line

Reply:
actually, it needs look like this: https://github.com/openshift/machine-config-operator/pull/1703/commits/db4a802817e7339da5ae74d751069e1d3e67614d
anoalog to the way it's done in installer

Reply:
I was mutating `files` there, reworked to return the array